core/execution/future/liquidation.go

```diff
@@ -0,0 +1,99 @@
+package future
+
+import (
+	"context"
+	"time"
+
+	"code.vegaprotocol.io/vega/core/events"
+	"code.vegaprotocol.io/vega/core/types"
+	"code.vegaprotocol.io/vega/libs/num"
+	"code.vegaprotocol.io/vega/logging"
+)
+
+func (m *Market) checkNetwork(ctx context.Context, now time.Time) error {
+	// debug
+	// this only returns an error if we couldn't get the price range, incidating no orders on book
+	order, _ := m.liquidation.OnTick(ctx, now)
+	if order == nil {
+		return nil
+	}
+	order.OriginalPrice = num.UintZero().Div(order.Price, m.priceFactor)
+	conf, err := m.matching.SubmitOrder(order)
+	if err != nil {
+		m.log.Panic("Failure after submitting order to matching engine",
+			logging.Order(order),
+			logging.Error(err),
+		)
+	}
+	// no order was submitted
+	if conf == nil {
+		return nil
+	}
+	m.broker.Send(events.NewOrderEvent(ctx, order))
+	m.handleConfirmationPassiveOrders(ctx, conf)
+
+	// no trades...
+	if len(conf.Trades) == 0 {
+		m.checkForReferenceMoves(ctx, conf.PassiveOrdersAffected, false)
+		return nil
+	}
```

Comment:
how come the references prices could have moved if there was not trade? I'm not sure what's the point of that? Also no passive order would have been affected?